internal/osbuild2/osap_remediation_stage.go

```diff
@@ -0,0 +1,46 @@
+package osbuild2
+
+type OscapRemediationStageOptions struct {
+	DataDir string      `json:"data_dir,omitempty"`
+	Config  OscapConfig `json:"config"`
+}
+
+func (OscapRemediationStageOptions) isStageOptions() {}
+
+func NewOscapRemediationStage(options *OscapRemediationStageOptions) *Stage {
+	return &Stage{
+		Type:    "org.osbuild.oscap.remediation",
+		Options: options,
+	}
+}
+
+func NewOscapRemediationStageOptions(dataDir *string, options OscapConfig) *OscapRemediationStageOptions {
+	return &OscapRemediationStageOptions{
+		DataDir: *dataDir,
+		Config: OscapConfig{
+			ProfileID:    options.ProfileID,
+			Datastream:   options.Datastream,
+			DatastreamID: options.DatastreamID,
+			XCCDFID:      options.XCCDFID,
+			BenchmarkID:  options.BenchmarkID,
+			ArfResult:    options.ArfResult,
+			HtmlReport:   options.HtmlReport,
+			VerboseLog:   options.VerboseLog,
+			VerboseLevel: options.VerboseLevel,
+		},
+	}
+}
+
+type OscapConfig struct {
+	Datastream   string `json:"datastream,omitempty" toml:"datastream,omitempty"`
+	ProfileID    string `json:"profile_id,omitempty" toml:"profile_id,omitempty"`
+	DatastreamID string `json:"datastream_id,omitempty" toml:"datastream_id,omitempty"`
+	XCCDFID      string `json:"xccdf_id,omitempty" toml:"xccdf_id,omitempty"`
+	BenchmarkID  string `json:"benchmark_id,omitempty" toml:"benchmark_id,omitempty"`
+	Tailoring    string `json:"tailoring,omitempty" toml:"tailoring,omitempty"`
+	TailoringID  string `json:"tailoring_id,omitempty" toml:"tailoring_id,omitempty"`
+	ArfResult    string `json:"arf_result,omitempty" toml:"arf_result,omitempty"`
+	HtmlReport   string `json:"html_report,omitempty" toml:"html_report,omitempty"`
+	VerboseLog   string `json:"verbose_log,omitempty" toml:"verbose_log,omitempty"`
+	VerboseLevel string `json:"verbose_level,omitempty" toml:"verbose_level,omitempty"`
```

Comment:
We could validate the values for this since it's an enum.
Have a look at the tar stage.
Validation like this can be hard (and annoying) to write and read, and we're not consistent about it, but I think we should try to add it for new stages.